File: backend/hostel-management/src/main/java/com/hostel/hostel_management/service/StudentService.java
package com.hostel.hostel_management.service;

import java.util.List;
import java.util.Optional;

import com.hostel.hostel_management.entity.Student;

public interface StudentService {
    Student createStudent(Student student);
    Student updateStudent(Long StudentId, Student student);
    void deleteStudent(Long StudentId);
    Student getStudentById(Long StudentId);
    List<Student> getAllStudents();
    Optional<Student> findByEnrollmentNo(String enrollmentNo);
    Student checkoutStudent(Long studentId);

    
    
    
}

File: frontend/hostel-frontend/src/services/studentService.js
import API from "../api/axios";

export const getStudents = () => {
  return API.get("/students");
};
export const getStudentById = (id) => {
  return API.get(`/students/${id}`);
};

export const createStudent = (student) => {
  return API.post("/students", student);
};

export const updateStudent = (id, student) => {
  return API.put(`/students/${id}`, student);
};

export const deleteStudent = (id) => {
  return API.delete(`/students/${id}`);
};

export const checkoutStudent = (id) => {
  return API.put(`/students/checkout/${id}`);
};